lib/pipeline.js

```diff
@@ -701,15 +703,26 @@ class PipelineModel extends BaseModel {
             // eslint-disable-next-line no-await-in-loop
             const permission = await user.getPermissions(this.scmUri);
 
-            if (permission.push) {
-                return user;
+            if (!permission.push) {
+                delete newAdmins[username];
             }
-
-            delete this.admins[username];
         }
         /* eslint-enable no-restricted-syntax */
 
-        throw new Error('Pipeline has no admin');
+        if (Object.keys(newAdmins).length === 0) {
+            throw new Error('Pipeline has no admin');
+        } else {
+            // This is needed to make admins dirty and update db
+            this.admins = newAdmins;
+            this.update();
```

Comment:
Just a question to make sure this is what we want. So we don't want to wait for this update to finish before we return?

Reply:
Thank you! I think I don't need to `update` at here. This function is called inside of `update`

https://github.com/screwdriver-cd/models/blob/master/lib/pipeline.js#L908